rubrics/tool/src/main/java/org/sakaiproject/rubrics/controller/GeneratePdfController.java

```diff
@@ -0,0 +1,268 @@
+    
+    @Autowired
+    private SessionManager sessionManager;
+
+    @Autowired
+    private PreferencesService preferencesService;
+
+    private static final String X_COPY_HEADER = "x-copy-source";
+    private static final String X_LANG_HEADER = "lang";
+    private static Font font = new Font(Font.FontFamily.HELVETICA, 10, Font.BOLD);
+    private static Font textFont = new Font(Font.FontFamily.HELVETICA, 8, Font.BOLD);
+    private static Font descFont = new Font(Font.FontFamily.HELVETICA, 7, Font.NORMAL);
+
+    @PreAuthorize("canCopy(#sourceId, 'Rubric')")
+    @RequestMapping(value = "/rubrics/generatePdf", headers = {X_COPY_HEADER,X_LANG_HEADER}) 
+    public ResponseEntity<StreamingResponseBody> generatePdf(@Param("sourceId") @RequestHeader(value = X_COPY_HEADER) String sourceId, 
+            @RequestHeader(value = X_LANG_HEADER) String lang, @RequestHeader(value = "toolId") String toolId,
+            @RequestHeader(value = "itemId") String entityId,
+            @RequestHeader(value = "evaluatedItemId") String evaluatedItemId)
+            throws CloneNotSupportedException {
+
+        Rubric sourceRubric = null;
+        String fileRubricName = null;
+        List<Evaluation> evals = null;
+        StreamingResponseBody stream = null;
+
+        try {
+            sourceRubric = rubricRepository.findById(Long.parseLong(sourceId))
+                .orElseGet(() -> rubricsConfiguration.getDefaultLayoutConfiguration(lang).getDefaultRubric());
+
+            evals = evaluationRepository.findByToolIdAndAssociationItemIdAndEvaluatedItemId(toolId, entityId, evaluatedItemId);
+            String fileName = sourceRubric.getTitle();
+            fileRubricName = fileName + ".pdf";
+            File tempFile = createPdf(sourceRubric, fileRubricName, evals);
+            Path pdfPath = Paths.get(tempFile.getAbsolutePath());
+            stream = outputStream -> {
+                Files.copy(pdfPath, outputStream);
+            };
+            //Files.deleteIfExists(pdfPath);
+        } catch (DocumentException | IOException ex) {
+            log.error(ex.getMessage(), ex);
+        }
+
+        HttpHeaders respHeaders = new HttpHeaders();
+        respHeaders.setContentType(new MediaType("application", "pdf"));
+        respHeaders.setContentDisposition( ContentDisposition.builder("attachment") .filename(fileRubricName).build());
+        return ResponseEntity.ok()
+                .headers(respHeaders)
+                .contentType(MediaType.parseMediaType("application/pdf"))
+                .body(stream);
+    }
+
+    public File createPdf(Rubric sourceRubric, String fileRubricName, List<Evaluation> evals)
+            throws DocumentException, IOException {
+
+        String userId = sessionManager.getCurrentSessionUserId();
+        final Locale locale = preferencesService.getLocale(userId);
+        // Count points
+        Integer points = 0;
+        if (null != evals && !evals.isEmpty()) {
```

Comment:
Does evaluationRepository.findByToolIdAndAssociationItemIdAndEvaluatedItemId return null or empty list when does not exist?

Reply:
Yes, if the rubric is blank and not graded.